lib/DirectoryWatcher/CMakeLists.txt

```diff
@@ -2,6 +2,8 @@ include(CheckIncludeFiles)
 
 set(LLVM_LINK_COMPONENTS support)
 
+CHECK_SYMBOL_EXISTS("FSEventStreamRef" "CoreServices/CoreServices.h" HAS_FS_EVENTS)
```

Comment:
In llvm it is called with the lowercased version, `check_symbol_exists`.
Also the macro will not be automatically be defined, you need to add a `#cmakedefine` in `include/clang/Config/config.h.cmake`.

For an example see how `HAVE_DLOPEN` is checked and defined in the llvm repo.